lib/narwin_chat/talks.ex

```diff
@@ -0,0 +1,103 @@
+defmodule NarwinChat.Talks do
+  require Logger
+
+  import Ecto.Query
+
+  alias NarwinChat.Repo
+  alias NarwinChat.Chat.Room
+  alias NarwinChat.Chat.Talk
+
+  def list_current_talks(opts \\ []) do
+    now = opts[:at] || DateTime.utc_now()
+
+    Talk
+    |> where([t], t.starts_at <= ^now and t.ends_at > ^now)
+    |> Repo.all()
+    |> Repo.preload([:room])
+  end
+
+  def update_talk_rooms(talks) do
+    for talk <- talks,
+        # Only update the room if the title is changing
+        false <- talk.title == talk.room.description do
+      talk.room
+      |> Room.changeset(%{description: talk.title})
+      |> Repo.update()
+      |> case do
+        {:ok, room} ->
+          Phoenix.PubSub.broadcast(NarwinChat.PubSub, "lobby", {:room_updated, room})
```

Comment:
I believe PubSub also needs to be added to the supervision tree

The broadcast could also be done via the Dispatcher, since that already sends messages to the lobby view, but either works

Reply:
Dispatcher probably does make more sense, to keep everything consistent. I honestly was kind of just rushing and wanted to get something working and I didn't have time to unravel what Dispatcher was doing, but I will take another look.

Reply:
Should be straightforward. Dispatcher has a list of pids for lobby listeners (https://github.com/liveviewnative/elixirconf_chat/blob/main/lib/narwin_chat/dispatcher.ex#L55), so you could add a cast that sends the new room to them.